pkg/sfu/receiver.go

```diff
@@ -202,10 +201,26 @@ func NewWebRTCReceiver(
 			return w.GetLayerDimension(quality)
 		},
 		GetMaxExpectedLayer: func() int32 {
-			return w.streamTrackerManager.GetMaxExpectedLayer()
+			// find min of <expected, published> layer
+			expectedLayer := w.streamTrackerManager.GetMaxExpectedLayer()
+			maxPublishedLayers := InvalidLayerSpatial
+			if w.TrackInfo != nil {
+				for _, layer := range w.TrackInfo.Layers {
+					if layer.Quality == livekit.VideoQuality_OFF {
+						continue
+					}
+					if int32(layer.Quality) > maxPublishedLayers {
+						maxPublishedLayers = int32(layer.Quality)
```

Comment:
There is this function - https://github.com/livekit/livekit-server/blob/0f7b9ae43ebf278d6bfb1dc4103d3c3f2d793a27/pkg/rtc/mediatrackreceiver.go#L401. Would be good to move that into `sfu/helpers` maybe and use it. Sorry for being a pain here. These are things that can get tangled up converting between different representations. So, want to have one definitive place for these and use it across the code base.